tilings/algorithms/gridded_perm_reduction.py

```diff
@@ -124,6 +125,28 @@ def minimal_reqs(
                 return (tuple(),)
         return tuple(minimized_requirements)
 
+    def remove_redundant(self, requirements: List[Requirement]) -> List[Requirement]:
+        relevant_reqs: List[Requirement] = []
+        for idx, requirement in enumerate(requirements):
+            if not all(requirement):
+                continue
+            if self._requirement_implied_by_some_requirement(
+                requirement,
+                chain(
+                    islice(requirements, idx),
+                    [
+                        gps
+                        for gps in islice(requirements, idx + 1, None)
+                        if gps != requirement
+                    ],
+                ),
```

Comment:
Is the use of islice and chain to keep this a generator? If so, should the second argument use () instead of [] so it too is a generator?

Reply:
Also why do we need to check gps != requirement? Equality would mean two identical requirements lists, and that should also be detected as redundant, right?

Reply:
it should be (), good catch.

the purpose of the `!=` is to ensure that if two are exactly the same, we leave one of them - the setup ensures we always leave the first one.

Reply:
Ah, very smart. Nice.